docs/docs/naming.md

```diff
@@ -56,6 +56,56 @@ Mesos-DNS generates A records for itself that list all the IP addresses that Mes
 
 In addition to A and SRV records for Mesos tasks, Mesos-DNS supports requests for SOA and NS records for the Mesos domain. DNS requests for records of other types in the Mesos domain will return `NXDOMAIN`. Mesos-DNS does not support PTR records needed fo reserve lookups. 
 
+## Customization of Records
+
+Mesos-DNS allows customization of the list of *record templates* that are used 
+to generate the A record names for each task. Compare the `Template` 
+configuration setting in the [configuration parameter documentation](configuration-parameters.md).
+
+By default only one *record template* is used:
+
+```
+{name}.{framework}
+```
+
+This gives the A records described above, e.g. `nginx.marathon.mesos.`.
+
+Additional templates can be defined using a number of *template variables*:
+
+- `framework`: the framework name, e.g. `marathon`
+- `slave-id-short`: the number in the slave ID after the last dash `-`, e.g. `0` for a slave ID `20140803-125133-3041283216-5050-2410-0`
+- `slave-id`: the complete slave ID, e.g. `20140803-125133-3041283216-5050-2410-0`
+- `task-id`: the complate task ID, e.g. `chronos.49b91a9a-3dda-11e4-a088-c20493233aa5`
+- `task-id-hash`: a hash value of the task ID, e.g. 76830, sufficiently unique for a realistic number of tasks on a cluster
+- `name`: the name of a task, provided by the framework either in the *TaskInfo* or the *DiscoveryInfo*, e.g. `nginx`
```

Comment:
assuming that the name in the DI trumps that of the task? if so, should clarify here